### PR TITLE
Monkey patch iris

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable) (N/A)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable) (N/A)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

### CODE REVIEW

**Comment by znicholls** [comment]
@lewisjared can you take a look. It seems ok except there must be some weird race condition as the scratch script is producing the following error (full log in email)

```
$ bash scripts/scratch-process.sh 
...
2019-07-10 10:44:59,394 ERROR:netcdf_scm.iris_cube_wrappers:Missing CF-netCDF measure variable 'areacella', referenced by netCDF variable 'sftlf'. Tried to add areacella cube but another exception was raised:
Traceback (most recent call last):
  File "...netcdf-scm/src/netcdf_scm/masks/__init__.py", line 387, in get_mask
    return self._masks[mask_name]
KeyError: 'World|Land'

ValueError: Duplicate cell_measures are not permitted.
```

File: src/netcdf_scm/masks/__init__.py [hunk 1]
```diff
@@ -111,6 +112,7 @@ def f(masker, cube, **kwargs):
     return f
 
 
+@lru_cache(maxsize=32)
```
Comment (lewisjared) [inline src/netcdf_scm/masks/__init__.py]: You would want to check if this is a per instance cache or if all instances use the same cache. If that is the case it doesn't do anything unless `get_default_sftlf_cube` is called multiple times per instance.

File: src/netcdf_scm/cli.py [hunk 2]
```diff
@@ -181,7 +181,7 @@ def crunch_files(fnames, dpath):
 
         tracker.register(out_filepath, scmcube.info)
         logger.info("Writing file to %s", out_filepath)
-        save_netcdf_scm_nc(results, out_filepath)
+        #save_netcdf_scm_nc(results, out_filepath)
```
Comment (lewisjared) [inline src/netcdf_scm/cli.py]: Is this meant to be in here?

File: src/netcdf_scm/iris_cube_wrappers.py [hunk 3]
```diff
@@ -721,8 +743,7 @@ def get_scm_cubes(self, sftlf_cube=None, land_mask_threshold=50):
             sftlf_cube=sftlf_cube, land_mask_threshold=land_mask_threshold
         )
         # force the data to realise so it's not read 9 times while applying masks
-        self.cube.data  # pylint:disable=pointless-statement
```
Comment (lewisjared) [inline src/netcdf_scm/iris_cube_wrappers.py]: Get rid of this if you have solved this issue

File: src/netcdf_scm/masks/__init__.py (hunk 1 above)
Comment (znicholls) [reply src/netcdf_scm/masks/__init__.py]: > If that is the case it doesn't do anything unless get_default_sftlf_cube is called multiple times per instance

It'll get called multiple times per instance so I think good enough?

Comment (lewisjared) [reply src/netcdf_scm/masks/__init__.py]: I was mistaken it is a module level function not a class member (Why does GH's first indent have to be so big).

File: scripts/scratch-process.sh [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 CONTACT='Zebedee Nicholls <zebedee.nicholls@climate-energy-college.org>, Jared Lewis <jared.lewis@climate-energy-college.org>, Malte Meinshausen <malte.meinshausen@unimelb.edu.au>'
 DRS="CMIP6Output"
-REGEXP="^(?!.*fx).*IPSL.*$"
+REGEXP="^(?!.*(fx|/ta/)).*(rlust|rsdt|rsut|tas|ts|fgco2|hfds|hfcorr|tos|co2|fco2fos|fco2nat|fco2antt|co2s).*$"
 # have to be super careful when crunching input files as the duplicate grids can cause
 # things to explode
 REGEXP_WRANGLE_IN_FILES=".*tas/.*gr/.*"
 
-netcdf-scm-crunch "${SRC_DIR}" "${CRUNCH_DIR}" "${CONTACT}"  --drs "${DRS}" --regexp "${REGEXP}"
+netcdf-scm-crunch "${SRC_DIR}" "${CRUNCH_DIR}" "${CONTACT}"  --drs "${DRS}" --regexp "${REGEXP}" --force
```
Comment (znicholls) [inline scripts/scratch-process.sh]:
```suggestion
netcdf-scm-crunch "${SRC_DIR}" "${CRUNCH_DIR}" "${CONTACT}"  --drs "${DRS}" --regexp "${REGEXP}" 
```

**Comment by rgieseke** [comment]
LGTM

image

**Comment by znicholls** [comment]
Closing in favour of #72

**Pull request closed without merging**